chromadb/telemetry/posthog.py

```diff
@@ -21,10 +22,29 @@ def __init__(self, system: System):
         posthog_logger = logging.getLogger("posthog")
         # Silence posthog's logging
         posthog_logger.disabled = True
+
+        self.batched_events: Dict[str, TelemetryEvent] = {}
+        self.seen_event_types: Set[Any] = set()
+
         super().__init__(system)
 
     @override
     def capture(self, event: TelemetryEvent) -> None:
+        if event.max_batch_size == 1 or event.batch_key not in self.seen_event_types:
```

Comment:
nit: I don't think you need seen_event_types anymore. self.batched_events should be sufficient for presence checking the batch_key?

Reply:
I believe we do: `self.batched_events` is used to keep track of the events currently being batched (grouped by `batch_key`). When an event is fully batched it's sent off and we `del self.batched_events[batch_key]`. `seen_event_types` is used to keep track of whether we've seen a `batch_key` at all because if we haven't we should fire the event immediately.

Reply:
Can't the batch size tell us that?

Reply:
I don't think so? If `batch_key not in self.batched_events` we don't know if we've already aggregated and fired off an event, or if we're seeing an event type for the first time

Reply:
Oh haha - of course. My bad!